CMake/kwiver-depends-VTK.cmake

```diff
@@ -6,16 +6,18 @@ option( KWIVER_ENABLE_VTK
   )
 
 if( KWIVER_ENABLE_VTK )
-    find_package(VTK REQUIRED
-        COMPONENTS
-        vtkCommonCore
-        vtkCommonDataModel
-        )
-    if(VTK_VERSION VERSION_LESS 8.2)
-        message(FATAL_ERROR "${PROJECT_NAME} supports VTK >= v8.2 "
-            "(Found ${VTK_VERSION})")
-    endif()
-
-  include(${VTK_USE_FILE})
-
+  find_package(VTK REQUIRED
+    COMPONENTS
+    vtkCommonCore
+    vtkCommonDataModel
+    )
+  if(VTK_VERSION VERSION_LESS 8.2)
+    message(FATAL_ERROR "${PROJECT_NAME} supports VTK >= v8.2 "
+      "(Found ${VTK_VERSION})")
+  elseif(NOT VTK_VERSION VERSION_LESS 8.90 )
+    get_target_property(VTK_INCLUDE_DIR VTK::CommonCore INTERFACE_INCLUDE_DIRECTORIES)
```

Comment:
Just curious, is `VTK::CommonCore` sufficient to get what we need regardless of what other VTK libraries may be added?  That is, is there always just one path we need for VTK_INCLUDE_DIR or could this be a list of paths that depends on which part of VTK we add?

Reply:
`VTK::CommonCore` should be sufficient since all headers are installed into `include/vtk-9.0/` and all includes in sub-directories from there have the path included when referenced. I could add a call for each module we list in COMPONENTS but I think it's superfluous. Sadly there isn't an actual VTK_INCLUDE_DIR anymore so we need to do it all through properties.

Reply:
Regarding `${VTK_LIBRARIES}` use, we could remove that, which is ok, it would just mean having an if/else dependent on VTK_VERSION.

Reply:
I am pushing an update for the `${VTK_LIBRARIES}` request. Should I expand the include search for all COMPONENTS listed in the find_package here as well? I can go either way, making several calls to `get_target_property` should be fast enough. I guess the only issue would come from adding a lot more VTK deps, but at least it's confined to a single file.

One other thought I had and I don't really know the answer .... If we do call get_target_property for all components, should we defer the include_dir search to the specific arrow that needs them? I guess the advantage is that if we are adding a lot of VTK deps, we can restrict the include search to what's required, and use target_include_directories which is a local include search. Thoughts?

Reply:
I don't understand.  I thought you said getting the include_dir for one target was sufficient since it is the same of all of them?

Reply:
> I don't understand. I thought you said getting the include_dir for one target was sufficient since it is the same of all of them?


It is ... I just wanted to confirm that you were ok with that. Sorry for the confusion.

Reply:
I'm okay with that, but @mwoehlke-kitware was implying there might be more correct way to do this.

Reply:
I assume that there is no `VTK_INCLUDE_DIRS` because it is expected to just link the libraries and get includes via `INTERFACE_INCLUDE_DIRS`. Okay, maybe that didn't work for VTK 8.x, but why can't we (continue to) `include(${VTK_USE_FILE})` if using VTK 8.x and do things the right way if using 9.x? My point was that `include_directories(SYSTEM "${VTK_INCLUDE_DIR}")` — indeed, any use of `include_directories` — seems dodgy.

Reply:
> I assume that there is no `VTK_INCLUDE_DIRS` because it is expected to just link the libraries and get includes via `INTERFACE_INCLUDE_DIRS`. Okay, maybe that didn't work for VTK 8.x, but why can't we (continue to) `include(${VTK_USE_FILE})` if using VTK 8.x and do things the right way if using 9.x? My point was that `include_directories(SYSTEM "${VTK_INCLUDE_DIR}")` — indeed, any use of `include_directories` — seems dodgy.

So, we are using `include(${VTK_USE_FILE})` for supported VTK versions < 8.90. I believe that version (I did check, btw) `VTK_USE_FILE` is deprecated and `VTK_INCLUDE_DIRS` is gone.

Using get_target_property requires a variable, so I used `VTK_INCLUDE_DIR`. I didn't know any other way to apply the result of the query without using include_directories. I think calling

` include ( ${VTK_DIR}/VTK-targets.cmake)`
`include ( ${VTK_DIR}/VTK-vtk-module-properties.cmake)`

is the right thing to do since it is stated in the VTK documents (though there is a lot of documentation there and it's not completely clear. Either way, doing those includes removes the need for `get_target_property` and `include_directories`, which puts us (close to) the same page at least.